lib/IDE/Refactoring.cpp

```diff
@@ -3887,7 +3889,1442 @@ bool RefactoringActionConvertToComputedProperty::performChange() {
+
+  ArrayRef<Expr *> args() { return Results.ref(); }
+};
+
+struct AsyncHandlerDesc {
+  const ParamDecl *Handler = nullptr;
+  int Index = -1;
+  HandlerResultType Type = HandlerResultType::INVALID;
+  bool HasError = false;
+
+  static AsyncHandlerDesc find(const CallExpr *CE, bool ignoreName = false) {
+    if (!CE)
+      return AsyncHandlerDesc();
+
+    auto *FD =
+        dyn_cast_or_null<FuncDecl>(CE->getFn()->getReferencedDecl().getDecl());
+    return find(FD, ignoreName);
+  }
+
+  static AsyncHandlerDesc find(const FuncDecl *FD, bool ignoreName = false) {
+    if (!FD || FD->hasAsync() || FD->hasThrows())
+      return AsyncHandlerDesc();
+
+    // Require at least one parameter and void return type
+    auto *Params = FD->getParameters();
+    if (Params->size() == 0 || !FD->getResultInterfaceType()->isVoid())
+      return AsyncHandlerDesc();
+
+    AsyncHandlerDesc HandlerDesc;
+
+    // Assume the handler is the last parameter for now
+    HandlerDesc.Index = Params->size() - 1;
+    HandlerDesc.Handler = Params->get(HandlerDesc.Index);
+    if (!ignoreName &&
+        !isCompletionHandlerName(HandlerDesc.Handler->getNameStr()))
+      return AsyncHandlerDesc();
+
+    // Callback must be a function type and return void. Doesn't need to have
+    // any parameters - may just be a "I'm done" callback
+    auto *HandlerTy = HandlerDesc.Handler->getType()->getAs<AnyFunctionType>();
+    if (!HandlerTy || !HandlerTy->getResult()->isVoid())
+      return AsyncHandlerDesc();
+
+    // Find the type of result in the handler (eg. whether it's a Result<...>,
+    // just parameters, or nothing).
+    auto HandlerParams = HandlerTy->getParams();
+    if (HandlerParams.size() == 1) {
+      auto ParamTy =
+          HandlerParams.back().getPlainType()->getAs<BoundGenericType>();
+      if (isResultType(ParamTy)) {
+        auto GenericArgs = ParamTy->getGenericArgs();
+        assert(GenericArgs.size() == 2 && "Result should have two params");
+        HandlerDesc.Type = HandlerResultType::RESULT;
+        HandlerDesc.HasError = !isNeverType(GenericArgs.back());
+      }
+    }
+
+    if (HandlerDesc.Type != HandlerResultType::RESULT) {
+      // Only handle non-result parameters
+      for (auto &Param : HandlerParams) {
+        if (isResultType(Param.getPlainType()))
+          return AsyncHandlerDesc();
+      }
+
+      HandlerDesc.Type = HandlerResultType::PARAMS;
+      if (!HandlerParams.empty()) {
+        auto LastParamTy = HandlerParams.back().getParameterType();
+        HandlerDesc.HasError =
+            isErrorType(LastParamTy->getOptionalObjectType());
+      }
+    }
+
+    return HandlerDesc;
+  }
+
+  bool isValid() const { return Type != HandlerResultType::INVALID; }
+
+  ArrayRef<AnyFunctionType::Param> params() const {
+    auto Ty = Handler->getType()->getAs<AnyFunctionType>();
+    assert(Ty && "Type must be a function type");
+    return Ty->getParams();
+  }
+
+  CallExpr *isCall(ASTNode Node) const {
```

Comment:
Sounds like it returns a bool at the moment. Maybe `getAsCall` or `getAsHandlerCall`?